beacon-chain/core/helpers/randao.go

```diff
@@ -4,35 +4,42 @@ import (
 	"fmt"
 
 	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
 	"github.com/prysmaticlabs/prysm/shared/hashutil"
 	"github.com/prysmaticlabs/prysm/shared/params"
 )
 
 // GenerateSeed generates the randao seed of a given epoch.
 //
 // Spec pseudocode definition:
-//   def generate_seed(state: BeaconState,
-//                  epoch: Epoch) -> Bytes32:
-//    """
-//    Generate a seed for the given ``epoch``.
-//    """
-//    return hash(
-//        get_randao_mix(state, epoch - MIN_SEED_LOOKAHED) +
-//        get_active_index_root(state, epoch)
-//    )
+// def generate_seed(state: BeaconState,
+// 	epoch: Epoch) -> Bytes32:
+// """
+// Generate a seed for the given ``epoch``.
+// """
+// return hash(
+// get_randao_mix(state, epoch - MIN_SEED_LOOKAHEAD) +
+// get_active_index_root(state, epoch) +
+// int_to_bytes32(epoch)
+// )
 func GenerateSeed(state *pb.BeaconState, wantedEpoch uint64) ([32]byte, error) {
-	if wantedEpoch > params.BeaconConfig().MinSeedLookahead {
-		wantedEpoch -= params.BeaconConfig().MinSeedLookahead
+	var wantedEpochM uint64
```

Comment:
What `wantedEpochM` means?

Reply:
+1

Can this just be something like this?

```go
if wantedEpoch >= params.BeaconConfig().MinSeedLookahead {
   return [32]byte{}, fmt.Errorf(...)
}
wantedEpoch -= params.BeaconConfig().MinSeedLookahead 
```

Reply:
it doesnt satisfy the paseudocode ```get_randao_mix(state, epoch - MIN_SEED_LOOKAHEAD)              + get_active_index_root(state, epoch) +...``` one time we use `epoch` one time we use 'epoch - MIN_SEED_LOOKAHEAD'

Reply:
will change the naming

Reply:
and use your method @prestonvanloon